gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -223,7 +223,7 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
             nonEmptyKspConfigurations.forEach {
```

Comment:
You should create a new configuration that extends from all non-empty ones. Something like
```
val processorClasspath = project.configurations.create(<unique_name>).extendsFrom(nonEmptyConfigurations)
kspTask.processorClasspath.from(processorClasspath)

class KspTask {
   @InputFiles
   @SkipWhenEmpty 
   abstract val processorClasspath: ConfigurableFileCollection // Gradle will instantiate this 
```

Reply:
Done. I removed `@SkipWhenEmtpy` because it seems to make Gradle treat the processorClasspath as source.

For example, when there is no source under `test/`, I think the desired behavior is always skipping the `kspTestKotlin` task, no matter whether there are processor classpath changes or not.

Also the annotations (`InputFiles` and `SkipWhenEmpty`) cannot be applied to `abstract val ...`, so I used `abstract fun getProcessorClasspath()` rather than `abstract val processorClasspath`

Reply:
SGTM. You can use `@get:InputFIles abstract val ...` to avoid creating the getter.

Reply:
So true! Updated.